src/pyOpenMS/pyTOPP/MRMMapper.py

```
@@ -8,56 +8,75 @@
                     if not allow_double_mappings: raise Exception("Cannot map twice")
                 mapped_already = True
-                precursor = chrom.getPrecursor();
-                peptide = targeted.getPeptideByRef(transition.getPeptideRef() )
-                precursor.setMetaValue("peptide_sequence", peptide.sequence)
+                precursor = chrom.getPrecursor()
+                
+                # Handle both peptides and nucleotides
+                if targeted.hasPeptide(transition.getTransGroupRef()):
+                    sequence_entity = targeted.getCompoundByRef(transition.getTransGroupRef())
+                elif targeted.hasNucleotide(transition.getTransGroupRef()):
+                    sequence_entity = targeted.getCompoundByRef(transition.getTransGroupRef())
```

Ruff (0.12.2)

Combine `if` branches using logical `or` operator

Combine `if` branches

(SIM114)